connector/countconnector/counter.go

```diff
@@ -41,9 +41,23 @@ func (c *counter[K]) update(ctx context.Context, attrs pcommon.Map, tCtx K) erro
 		countAttrs := pcommon.NewMap()
 		for _, attr := range md.attrs {
 			if attrVal, ok := attrs.Get(attr.Key); ok {
-				countAttrs.PutStr(attr.Key, attrVal.Str())
-			} else if attr.DefaultValue != "" {
-				countAttrs.PutStr(attr.Key, attr.DefaultValue)
+				switch typeAttr := attrVal.Type(); typeAttr {
+				case pcommon.ValueTypeInt:
+					countAttrs.PutInt(attr.Key, attrVal.Int())
+				case pcommon.ValueTypeDouble:
+					countAttrs.PutDouble(attr.Key, attrVal.Double())
+				default:
+					countAttrs.PutStr(attr.Key, attrVal.Str())
+				}
+			} else {
+				switch v := attr.DefaultValue.(type) {
```

Comment:
@djaglowski Thanks for review.  I agree that your idea simplifies usage. I just have a problem with this part of the code, I'm not sure if I need to match for each type int -> `int, int8, int16, int32, int64` and the same doubt for float.

Reply:
Sure, it's verbose but doable. [Here's a decently comprehensive list](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/11d29882ebd3c97c839a590fcba089e27de6b355/pkg/stanza/adapter/converter.go#L285) which is used elsewhere.

Reply:
Hmm, good point. I suppose since we know these values can only come from configuration, we may not need to handle all types of ints, etc. I'm not sure how we establish guarantees around this but I'm also comfortable just getting a reasonable first implementation in place and adjusting it later if needed.

Reply:
Thanks, that leaves it as it is. PR is ready for review.